### ISSUE TITLE
pinadd isn't getting URL for me (Safari on OS X 10.10.4)

### ISSUE THREAD

**Issue by kbradnam** [vitorgalvao/alfred-workflows#15]
I could get pinadd configured with my Pinboard API token, but every time I run the workflow it fails to add anything. I hacked the pinadd script to print out the value of `${url}` and it is empty for me. So on my browser (latest version of safari) it just doesn't seem to grab anything for the frontmost URL.

**Comment by vitorgalvao** [comment]
Thank you for doing some of the debugging work, already. I have a suspicion of what might be the problem. Open the workflow’s folder in the Finder (right-click on its name on the workflows list in Alfred and pick `Show in Finder`, and try to open the `setTitleAndURL` executable. Does it open, or does it show the “can’t open because it’s from an unidentified developer message”? If it shows the message, right-click the executable and chose `Open`. Let it run that once (it’ll open a terminal) and it should work from then on.

**Comment by kbradnam** [comment]
On Wed, Jul 1, 2015, at 03:35 PM, Vítor Galvão wrote:
> Thank you for doing some of the debugging work, already. I have a
> suspicion of what might be the problem. Open the workflow’s folder in
> the Finder (right-click on its name on the workflows list in Alfred
> and pick Show in Finder, and try to open the setTitleAndURL
> executable. Does it open, or does it show the “can’t open because
> it’s from an unidentified developer message”? If it shows the
> message, right-click the executable and chose Open. Let it run that
> once (it’ll open a


Okay. I did all of this, and then ran into the next permissions
issue with the terminal-notifier executable. I then granted (via the
pop up window that appeared) permission to open this script, but I
still get the error message about adding the bookmark. But there are
no further warnings.

I can confirm that in the terminal I can run something like the
following without problem:

./_licensed/terminal-notifier/terminal-
notifier.app/Contents/MacOS/terminal-notifier -message test

Regards,

Keith

**Comment by vitorgalvao** [comment]
But is `${url}` still empty?

**Comment by kbradnam** [comment]
On Wed, Jul 1, 2015, at 03:48 PM, Vítor Galvão wrote:
> But is ${url} still empty?


> — Reply to this email directly or view it on GitHub[1].



Yes. If I add the following line to pinadd:

notification "URL is ${URL}."

And then try calling the workflow, I quickly see the notification appear
with 'URL is .' and then it moves on to the error message notification.

Regards,

Keith


--
Keith Bradnam
 
+ twitter: kbradnam
+ web: http://keithbradnam.com and http://acgt.me
+
+ Q: Why is this email five sentences or less? A: http://five.sentenc.es
 


Links:

  1. https://github.com/vitorgalvao/alfred-workflows/issues/15#issuecomment-117844318

**Comment by vitorgalvao** [comment]
What was the command you used to configure the workflow? Did you `configurepinadd username:HSJWJK2HHSKI14QPDOIK safari`? Just to make sure you did not forget to pick a browser.

**Comment by kbradnam** [comment]
I've solved it.

Short story: I did forget to include the browser name when configuring

Longer story: I did try correcting that at one point, but I realized
that when I did then I omitted my Pinboard account name from the
configuration details (gave just the API token).

My bad. Perhaps it would help if the configure step reported a
notification of what exactly went missing? That would have saved me (and
probably yourself) some time in troubleshooting.

Apologies for troubling you about this! I look forward to using PinAdd!

Regards,

Keith


On Thu, Jul 2, 2015, at 04:34 AM, Vítor Galvão wrote:
> What was the command you used to configure the workflow? Did you
> configurepinadd username:HSJWJK2HHSKI14QPDOIK safari? Just to make
> sure you did not forget to pick a browser.


> — Reply to this email directly or view it on GitHub[1].


>

--
Keith Bradnam
 
+ twitter: kbradnam
+ web: http://keithbradnam.com and http://acgt.me
+
+ Q: Why is this email five sentences or less? A: http://five.sentenc.es
 


Links:

  1. https://github.com/vitorgalvao/alfred-workflows/issues/15#issuecomment-118005446

**Comment by vitorgalvao** [comment]
Yes, that could be a bit better, and I’ve long thought about fixing it. PinAdd was actually one of the very first workflows I did, so I was learning Alfred’s capabilities and workflow interactions as I went. As a result, not every corner of it is as polished as some of the later ones. There are two main reasons why I haven’t yet fixed that interaction, though the workflow itself as a whole as suffered many major changes:

1. It is a low priority feature (in the sense you only ever do it once).
2. I have plans to eventually (no set date) introduce new functionality I’ve been wanting for a while, and I should revamp everything when (if) that happens. I’ll probably take that as an opportunity to play with Swift.

> Apologies for troubling you about this! I look forward to using PinAdd!

No trouble at all. I hope you like it.
